src/util/FastMap.ts

```diff
@@ -1,23 +1,31 @@
 export class FastMap {
-  size: number = 0;
-  private _values: Object = {};
+  private _size: number = 0;
+  get size(): number {
+    return this._size;
+  }
+  private values: Object = {};
 
   delete(key: string): boolean {
-    this._values[key] = null;
+    this.values[key] = null;
+    this._size--;
     return true;
   }
 
   set(key: string, value: any): FastMap {
-    this._values[key] = value;
+    if (!this.get(key)) {
```

Comment:
If we're not using `size` we can just get rid of it. There's no reason to assert and increment if we're not even using the value we're incrementing. Extra overhead for no reason.